### ISSUE TITLE
New ocf-data-sampler models not performing

### ISSUE THREAD

**Issue by peterdudfield** [openclimatefix/uk-pvnet-app#222]
It looks like there is a bug with the new ocf-data-sampler models

My guess is it something to do with the satellite data

![Image](https://github.com/user-attachments/assets/d46c45b5-fd91-4e98-ba2f-abe0abc1eebd)

@felix-e-h-p can you check that the models were trained with satellite normalizing. 
Might be worth checking a sample that is was trained on, and seeing what the satellite values are. 

This is what the satellite values in the sample look like, this morning
![Image](https://github.com/user-attachments/assets/deece204-bf4a-4260-9e7e-7ce16af55d27)

**Comment by peterdudfield** [comment]
It could be in training they were normzalised differently, or not at all?

**Comment by felix-e-h-p** [comment]
Few example samples around 7AM:

Satellite Data Inspection: 00570568.pt

1. Satellite Time UTC:
  Time 0: 2021-08-05 07:30:00 UTC
  Time 1: 2021-08-05 07:35:00 UTC
  Time 2: 2021-08-05 07:40:00 UTC
  Time 3: 2021-08-05 07:45:00 UTC
  Time 4: 2021-08-05 07:50:00 UTC
  Time 5: 2021-08-05 07:55:00 UTC
  Time 6: 2021-08-05 08:00:00 UTC
  Time 7: 2021-08-05 08:05:00 UTC
  Time 8: 2021-08-05 08:10:00 UTC
  Time 9: 2021-08-05 08:15:00 UTC
  Time 10: 2021-08-05 08:20:00 UTC
  Time 11: 2021-08-05 08:25:00 UTC
  Time 12: 2021-08-05 08:30:00 UTC
  Time 13: 2021-08-05 08:35:00 UTC
  Time 14: 2021-08-05 08:40:00 UTC
  Time 15: 2021-08-05 08:45:00 UTC
  Time 16: 2021-08-05 08:50:00 UTC
  Time 17: 2021-08-05 08:55:00 UTC
  Time 18: 2021-08-05 09:00:00 UTC

2. Satellite Actual Data:
  Shape: torch.Size([19, 11, 24, 24])
  Data type: torch.float16
  Min value: 0.047515869140625
  Max value: 0.953125
  Mean value: 0.6069952845573425

3. Satellite Actual Data Detailed:
  Time step 0:
    Shape: torch.Size([11, 24, 24])
    Min: 0.054779052734375
    Max: 0.94189453125
    Mean: 0.587161660194397
  Time step 1:
    Shape: torch.Size([11, 24, 24])
    Min: 0.0513916015625
    Max: 0.94384765625
    Mean: 0.5877909064292908
  Time step 2:
    Shape: torch.Size([11, 24, 24])
    Min: 0.05596923828125
    Max: 0.94580078125
    Mean: 0.5918352007865906
  Time step 3:
    Shape: torch.Size([11, 24, 24])
    Min: 0.047515869140625
    Max: 0.94921875
    Mean: 0.5946084856987
  Time step 4:
    Shape: torch.Size([11, 24, 24])
    Min: 0.054046630859375
    Max: 0.947265625
    Mean: 0.5969992876052856

4. Satellite Geostationary Coordinates:
  X coordinates: tensor([-792106.4358, -789106.0326, -786105.6294, -783105.2263, -780104.8231,
        -777104.4199, -774104.0168, -771103.6136, -768103.2104, -765102.8073,
        -762102.4041, -759102.0010, -756101.5978, -753101.1946, -750100.7915,
        -747100.3883, -744099.9851, -741099.5820, -738099.1788, -735098.7756,
        -732098.3725, -729097.9693, -726097.5661, -723097.1630],
       dtype=torch.float64)
  Y coordinates: tensor([4806645.8716, 4809646.2748, 4812646.6780, 4815647.0811, 4818647.4843,
        4821647.8875, 4824648.2906, 4827648.6938, 4830649.0970, 4833649.5001,
        4836649.9033, 4839650.3065, 4842650.7096, 4845651.1128, 4848651.5160,
        4851651.9191, 4854652.3223, 4857652.7255, 4860653.1286, 4863653.5318,
        4866653.9350, 4869654.3381, 4872654.7413, 4875655.1445],
       dtype=torch.float64)

==================================================

Satellite Data Inspection: 00484827.pt

1. Satellite Time UTC:
  Time 0: 2018-03-04 07:30:00 UTC
  Time 1: 2018-03-04 07:35:00 UTC
  Time 2: 2018-03-04 07:40:00 UTC
  Time 3: 2018-03-04 07:45:00 UTC
  Time 4: 2018-03-04 07:50:00 UTC
  Time 5: 2018-03-04 07:55:00 UTC
  Time 6: 2018-03-04 08:00:00 UTC
  Time 7: 2018-03-04 08:05:00 UTC
  Time 8: 2018-03-04 08:10:00 UTC
  Time 9: 2018-03-04 08:15:00 UTC
  Time 10: 2018-03-04 08:20:00 UTC
  Time 11: 2018-03-04 08:25:00 UTC
  Time 12: 2018-03-04 08:30:00 UTC
  Time 13: 2018-03-04 08:35:00 UTC
  Time 14: 2018-03-04 08:40:00 UTC
  Time 15: 2018-03-04 08:45:00 UTC
  Time 16: 2018-03-04 08:50:00 UTC
  Time 17: 2018-03-04 08:55:00 UTC
  Time 18: 2018-03-04 09:00:00 UTC

2. Satellite Actual Data:
  Shape: torch.Size([19, 11, 24, 24])
  Data type: torch.float16
  Min value: 0.0
  Max value: 0.0
  Mean value: 0.0

3. Satellite Actual Data Detailed:
  Time step 0:
    Shape: torch.Size([11, 24, 24])
    Min: 0.0
    Max: 0.0
    Mean: 0.0
  Time step 1:
    Shape: torch.Size([11, 24, 24])
    Min: 0.0
    Max: 0.0
    Mean: 0.0
  Time step 2:
    Shape: torch.Size([11, 24, 24])
    Min: 0.0
    Max: 0.0
    Mean: 0.0
  Time step 3:
    Shape: torch.Size([11, 24, 24])
    Min: 0.0
    Max: 0.0
    Mean: 0.0
  Time step 4:
    Shape: torch.Size([11, 24, 24])
    Min: 0.0
    Max: 0.0
    Mean: 0.0

4. Satellite Geostationary Coordinates:
  X coordinates: tensor([-684091.9218, -681091.5186, -678091.1155, -675090.7123, -672090.3091,
        -669089.9060, -666089.5028, -663089.0996, -660088.6965, -657088.2933,
        -654087.8901, -651087.4870, -648087.0838, -645086.6807, -642086.2775,
        -639085.8743, -636085.4712, -633085.0680, -630084.6648, -627084.2617,
        -624083.8585, -621083.4553, -618083.0522, -615082.6490],
       dtype=torch.float64)
  Y coordinates: tensor([4854652.3223, 4857652.7255, 4860653.1286, 4863653.5318, 4866653.9350,
        4869654.3381, 4872654.7413, 4875655.1445, 4878655.5476, 4881655.9508,
        4884656.3540, 4887656.7571, 4890657.1603, 4893657.5634, 4896657.9666,
        4899658.3698, 4902658.7729, 4905659.1761, 4908659.5793, 4911659.9824,
        4914660.3856, 4917660.7888, 4920661.1919, 4923661.5951],
       dtype=torch.float64)

==================================================

Inspecting Validation Set Satellite Data:
Satellite Data Inspection: 00005709.pt

1. Satellite Time UTC:
  Time 0: 2023-03-05 06:30:00 UTC
  Time 1: 2023-03-05 06:35:00 UTC
  Time 2: 2023-03-05 06:40:00 UTC
  Time 3: 2023-03-05 06:45:00 UTC
  Time 4: 2023-03-05 06:50:00 UTC
  Time 5: 2023-03-05 06:55:00 UTC
  Time 6: 2023-03-05 07:00:00 UTC
  Time 7: 2023-03-05 07:05:00 UTC
  Time 8: 2023-03-05 07:10:00 UTC
  Time 9: 2023-03-05 07:15:00 UTC
  Time 10: 2023-03-05 07:20:00 UTC
  Time 11: 2023-03-05 07:25:00 UTC
  Time 12: 2023-03-05 07:30:00 UTC
  Time 13: 2023-03-05 07:35:00 UTC
  Time 14: 2023-03-05 07:40:00 UTC
  Time 15: 2023-03-05 07:45:00 UTC
  Time 16: 2023-03-05 07:50:00 UTC
  Time 17: 2023-03-05 07:55:00 UTC
  Time 18: 2023-03-05 08:00:00 UTC

2. Satellite Actual Data:
  Shape: torch.Size([19, 11, 24, 24])
  Data type: torch.float16
  Min value: 0.01160430908203125
  Max value: 0.908203125
  Mean value: 0.5192838907241821

3. Satellite Actual Data Detailed:
  Time step 0:
    Shape: torch.Size([11, 24, 24])
    Min: 0.01160430908203125
    Max: 0.908203125
    Mean: 0.5083804130554199
  Time step 1:
    Shape: torch.Size([11, 24, 24])
    Min: 0.01268768310546875
    Max: 0.9072265625
    Mean: 0.5077431797981262
  Time step 2:
    Shape: torch.Size([11, 24, 24])
    Min: 0.01377105712890625
    Max: 0.90771484375
    Mean: 0.507448136806488
  Time step 3:
    Shape: torch.Size([11, 24, 24])
    Min: 0.0159454345703125
    Max: 0.9072265625
    Mean: 0.5077803134918213
  Time step 4:
    Shape: torch.Size([11, 24, 24])
    Min: 0.019195556640625
    Max: 0.90771484375
    Mean: 0.508722722530365

4. Satellite Geostationary Coordinates:
  X coordinates: tensor([-651087.4870, -648087.0838, -645086.6807, -642086.2775, -639085.8743,
        -636085.4712, -633085.0680, -630084.6648, -627084.2617, -624083.8585,
        -621083.4553, -618083.0522, -615082.6490, -612082.2458, -609081.8427,
        -606081.4395, -603081.0363, -600080.6332, -597080.2300, -594079.8268,
        -591079.4237, -588079.0205, -585078.6173, -582078.2142],
       dtype=torch.float64)
  Y coordinates: tensor([4596617.6500, 4599618.0532, 4602618.4564, 4605618.8595, 4608619.2627,
        4611619.6659, 4614620.0690, 4617620.4722, 4620620.8754, 4623621.2785,
        4626621.6817, 4629622.0849, 4632622.4880, 4635622.8912, 4638623.2944,
        4641623.6975, 4644624.1007, 4647624.5039, 4650624.9070, 4653625.3102,
        4656625.7133, 4659626.1165, 4662626.5197, 4665626.9228],
       dtype=torch.float64)
Satellite Data Inspection: 00087379.pt

1. Satellite Time UTC:
  Time 0: 2022-05-11 07:30:00 UTC
  Time 1: 2022-05-11 07:35:00 UTC
  Time 2: 2022-05-11 07:40:00 UTC
  Time 3: 2022-05-11 07:45:00 UTC
  Time 4: 2022-05-11 07:50:00 UTC
  Time 5: 2022-05-11 07:55:00 UTC
  Time 6: 2022-05-11 08:00:00 UTC
  Time 7: 2022-05-11 08:05:00 UTC
  Time 8: 2022-05-11 08:10:00 UTC
  Time 9: 2022-05-11 08:15:00 UTC
  Time 10: 2022-05-11 08:20:00 UTC
  Time 11: 2022-05-11 08:25:00 UTC
  Time 12: 2022-05-11 08:30:00 UTC
  Time 13: 2022-05-11 08:35:00 UTC
  Time 14: 2022-05-11 08:40:00 UTC
  Time 15: 2022-05-11 08:45:00 UTC
  Time 16: 2022-05-11 08:50:00 UTC
  Time 17: 2022-05-11 08:55:00 UTC
  Time 18: 2022-05-11 09:00:00 UTC

2. Satellite Actual Data:
  Shape: torch.Size([19, 11, 24, 24])
  Data type: torch.float16
  Min value: 0.057525634765625
  Max value: 0.92529296875
  Mean value: 0.6108161211013794

3. Satellite Actual Data Detailed:
  Time step 0:
    Shape: torch.Size([11, 24, 24])
    Min: 0.0789794921875
    Max: 0.9189453125
    Mean: 0.5921483039855957
  Time step 1:
    Shape: torch.Size([11, 24, 24])
    Min: 0.07061767578125
    Max: 0.921875
    Mean: 0.5939205288887024
  Time step 2:
    Shape: torch.Size([11, 24, 24])
    Min: 0.08160400390625
    Max: 0.92236328125
    Mean: 0.5957145690917969
  Time step 3:
    Shape: torch.Size([11, 24, 24])
    Min: 0.08746337890625
    Max: 0.92236328125
    Mean: 0.5974012613296509
  Time step 4:
    Shape: torch.Size([11, 24, 24])
    Min: 0.0855712890625
    Max: 0.92138671875
    Mean: 0.5999493598937988

4. Satellite Geostationary Coordinates:
  X coordinates: tensor([-879118.1276, -876117.7244, -873117.3213, -870116.9181, -867116.5149,
        -864116.1118, -861115.7086, -858115.3054, -855114.9023, -852114.4991,
        -849114.0959, -846113.6928, -843113.2896, -840112.8864, -837112.4833,
        -834112.0801, -831111.6769, -828111.2738, -825110.8706, -822110.4674,
        -819110.0643, -816109.6611, -813109.2579, -810108.8548],
       dtype=torch.float64)
  Y coordinates: tensor([4857652.7255, 4860653.1286, 4863653.5318, 4866653.9350, 4869654.3381,
        4872654.7413, 4875655.1445, 4878655.5476, 4881655.9508, 4884656.3540,
        4887656.7571, 4890657.1603, 4893657.5634, 4896657.9666, 4899658.3698,
        4902658.7729, 4905659.1761, 4908659.5793, 4911659.9824, 4914660.3856,
        4917660.7888, 4920661.1919, 4923661.5951, 4926661.9983],
       dtype=torch.float64)

**Comment by dfulu** [comment]
@felix-e-h-p you may have noticed here that all the satellite values in `00484827.pt` are zero. The entirety of our 2018 satellite data is NaNs so I haven't been using it in my batches. It could be that the model could learn to deal with that satellite dropout, and the extra year might help train the NWP encoder. But it has felt cleaner to me not to use this data
